util/oteltest/README.md

```diff
@@ -0,0 +1,194 @@
+NUM_ADDS = 12
+
+if __name__ == "__main__":
+    tracer = trace.get_tracer("my-tracer")
+    for i in range(NUM_ADDS):
+        with tracer.start_as_current_span("my-span"):
+            print(f"simple_loop.py: {i+1}/{NUM_ADDS}")
+            time.sleep(0.5)
+
+
+class MyOtelTest(OtelTest):
+    def requirements(self):
+        return "opentelemetry-distro", "opentelemetry-exporter-otlp-proto-grpc"
+
+    def environment_variables(self):
+        return {"OTEL_SERVICE_NAME": SERVICE_NAME}
+
+    def wrapper_command(self):
+        return "opentelemetry-instrument"
+
+    def on_start(self):
+        return None
+
+    def on_stop(self, telemetry: Telemetry, stdout: str, stderr: str, returncode: int) -> None:
+        assert telemetry.num_spans() == NUM_ADDS
+```
+
+Here's a client-server example:
+
+```python
+import time
+from typing import Mapping, Optional, Sequence
+
+from flask import Flask, jsonify
+
+from oteltest import OtelTest, Telemetry
+
+PORT = 8002
+HOST = "127.0.0.1"
+
+
+app = Flask(__name__)
+
+
+@app.route("/")
+def home():
+    return jsonify({"library": "flask"})
+
+
+if __name__ == "__main__":
+    app.run(port=PORT, host=HOST)
+
+
+class FlaskTest(OtelTest):
+    def environment_variables(self) -> Mapping[str, str]:
+        return {}
+
+    def requirements(self) -> Sequence[str]:
+        return (
+            "opentelemetry-distro",
+            "opentelemetry-exporter-otlp-proto-grpc",
+            "opentelemetry-instrumentation-flask",
+            "flask==2.3.3",
+            "jsonify",
+        )
+
+    def wrapper_command(self) -> str:
+        return "opentelemetry-instrument"
+
+    def on_start(self) -> Optional[float]:
+        import http.client
+
+        # Todo: replace this sleep with a liveness check!
+        time.sleep(10)
+
+        conn = http.client.HTTPConnection(HOST, PORT)
+        conn.request("GET", "/")
+        print("response:", conn.getresponse().read().decode())
+        conn.close()
+
+        # The return value of on_script_start() tells oteltest the number of seconds to wait for the script to complete.
+        # In this case, we indicate 30 (seconds), which, once elapsed, will cause the script to be terminated, if it's
+        # still running. If we return `None` then the script will run indefinitely.
+        return 30
+
+    def on_stop(self, telemetry: Telemetry, stdout: str, stderr: str, returncode: int) -> None:
+        print(f"on_stop: telemetry: {telemetry}")
+
+```
+
+### otelsink
+
+`otelsink` is a gRPC server that listens for OTel metrics, traces, and logs.
+
+#### Operation
+
+You can run otelsink either from the command line by using the `otelsink` command (installed when you
+`pip install oteltest`), or programmatically.
+
+Either way, `otelsink` runs a gRPC server listening on 0.0.0.0:4317.
+
+#### Command Line
+
+```
+% otelsink
+starting otelsink with print handler
+```
+
+#### Programmatic
+
+```
+from oteltest.sink import GrpcSink, PrintHandler
```

Comment:
Is `PrintHandler` import needed for this example?

```suggestion
from oteltest.sink import GrpcSink, RequestHandler
```

Reply:
Good catch 😎. Fixed.